app/services/schools_in_area.rb

```diff
@@ -0,0 +1,45 @@
+class SchoolsInArea
+  attr_accessor :lat, :lng, :radius
+
+  def initialize(lat:, lng:, radius:)
+    self.lat = lat
+    self.lng = lng
+    self.radius = radius
+  end
+
+  def results
+    schools.select do |s|
+      place = data.find { |d| d.destination == s.place }
+      distance_in_miles(place.distance_in_meters) <= radius
+    end
+  end
+
+  private
+
+  def distance_in_miles(metres)
+    metres * 0.00062137
+  end
+
+  def schools
+    @schools ||= School.within(radius, origin: [lat, lng])
+                       .joins(:vacancies)
+                       .merge(Vacancy.live)
+                       .distinct
+                       .order(:id)
+  end
+
+  def matrix
+    @matrix ||= begin
+      matrix = GoogleDistanceMatrix::Matrix.new
+      matrix.origins << GoogleDistanceMatrix::Place.new(lat: lat, lng: lng)
+      schools.each do |s|
+        matrix.destinations << s.place
+      end
+      matrix
```

Comment:
Do we need to guard against there being more than 25? Perhaps log something so we hear about it, and fall back to a basic radius search? Or make more API calls?

Reply:
Yep, I've done this now - breaking them down into batches of 25 and making subsequent calls. I'm still curious / concerned about how this will work in practice. Are we going to end up with really poor performance on areas with lots of schools?

Reply:
If we log something with the number of schools, or some note about having to batch them, maybe that would help us track if it's occuring at least?